### PR TITLE
gst-plugins-rs 0.9.3

### PR DESCRIPTION
**Pull request by mistydemeo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This version introduces a hard dependency on the webrtc plugin, which means it needs another one of the gst-plugin packs as a dependency. It also requires Python 3.11 for its buildsystem.

Closes #115929.

### CODE REVIEW

**Comment by mistydemeo** [comment]
Linux-only build error is going to be hard for me to diagnose. :/

**Comment by cho-m** [comment]
Script may be using Python 3.10 rather than Python 3.11. `PATH` has `/home/linuxbrew/.linuxbrew/opt/python@3.10/bin` before `/home/linuxbrew/.linuxbrew/opt/python@3.11/libexec/bin` so `#!/usr/bin/env python3` would probably pick up Python 3.10 first.

File: Formula/gst-plugins-rs.rb [hunk 1]
```diff
@@ -20,18 +19,19 @@ class GstPluginsRs < Formula
...
   depends_on "gtk4"
   depends_on "pango" # for closedcaption
 
-  # commit ref, https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/commit/ea98a0b5964cd196abbb48c621969a8ef33eb157
-  # remove in next release
-  patch :DATA
-
   def install
+    # Fixes an issue where the wrong Python is picked up in the Linux build
+    ENV.prepend_path "PATH", Formula["python@3.11"].opt_bin
```
Comment (cho-m) [inline Formula/gst-plugins-rs.rb:33]: We no longer provide `python3` in `bin` for non-aliased Pythons.
```suggestion
    ENV.prepend_path "PATH", Formula["python@3.11"].opt_libexec/"bin"
```

**Comment by cho-m** [comment]
The Intel failure is SSE4.1-related due to missing some source files when a Rust dependency builds a copy of `webp` manually https://github.com/qnighy/libwebp-sys2-rs/blob/v0.1.4/build.rs#L85-L86 

Specifically, build has `c_src/src/dsp/lossless.c` and `c_src/src/dsp/lossless_sse2.c` but not `c_src/src/dsp/lossless_sse41.c`.

Currently doesn't support system `webp` on macOS, but we might be able to use it on Linux.

---

It does look like `gst-plugins-rs` was testing/using an older version of this library (v0.1.2) but dependency resolution allows newer version since they don't include a Cargo.lock and their Cargo.toml allows it. The other version had an older libwebp without the extra SSE4.1 file.

File: Formula/gst-plugins-rs.rb [hunk 2]
```diff
@@ -20,18 +24,53 @@ class GstPluginsRs < Formula
...
+        end
+      end
+    end
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    # TODO: Use `webp` dependency on macOS when supported.
+    # Issue ref: https://github.com/qnighy/libwebp-sys2-rs/issues/4
+    depends_on "webp"
```
Comment (cho-m) [inline Formula/gst-plugins-rs.rb:62]: Need to check if library is linked to make sure build picked up Homebrew's copy.

**Comment by sdroege** [comment]
> It also requires Python 3.11 for its buildsystem.

I would recommend using `cargo-c` instead of the meson-based build system if you can. One layer of build systems less.

If you want to use the meson-based build system, then it also works with older Python versions if you can ensure that the `tomli` module is available.

> It does look like gst-plugins-rs was testing/using an older version of this library (v0.1.2) but dependency resolution allows newer version since they don't include a Cargo.lock and their Cargo.toml allows it.

The [release branches](https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/tree/0.9) and [tags](https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/tree/0.9.3) include a `Cargo.lock`. That includes `libwebp2-sys` 0.1.4 though.

In GStreamer's macOS/Windows binaries the webp plugin is currently not included (someone would have to add it and libwebp), which is why this wasn't noticed there.

**Pull request closed without merging**